src/raiden_libs/matrix.py

```diff
@@ -356,10 +355,7 @@ def start(self, user_manager: MultiClientUserAddressManager) -> None:
         self.user_manager = user_manager
         try:
             self._start_client(self.main_client.api.base_url)
-        except (TransportError, ConnectionError):
-            # When the sync worker fails, waiting for startup_finished does not
-            # make any sense.
-            self.startup_finished.set()
+        except (TransportError, ConnectionError, MatrixRequestError):
             return
```

Comment:
Is this correct?
If we can't connect just silently return?

Hm, I assume the assert in `MatrixListener._run()` will fail. But then don't we deadlock on the `joinall` on the `startup_finished_greenlet`?

Reply:
It will actually be handled in `MatrixListener._run()`. The main client is different from the other clients. If the main client fails for whatever reason the PFS shuts down and restarts. The forever sync loop only exists for the other clients.

When it silently returns the run method will end. the joinall has a timeout set to 0 and thus should not dead lock should it?

Reply:
I added a log for better verbosity

Reply:
> It will actually be handled in MatrixListener._run()

Yeah I got that. But it feels a bit convoluted.

> the joinall has a timeout set to 0 and thus should not dead lock should it?

Ah right. 
Well it's not related to this PR so let's merge it.